creds/kubernetes/kubernetes.go

```diff
@@ -23,22 +23,15 @@ func (k Kubernetes) Get(varDef template.VariableDefinition) (interface{}, bool,
 	var pipelineSecretName = k.PipelineName + "." + varDef.Name
 	var secretName = varDef.Name
 
-	secret, found, err := k.findSecret(namespace, pipelineSecretName)
```

Comment:
I preferred the code before; `foo, found, err` is more explicit. I think we should explicitly check for `403` instead.

By not checking the type of error, this could result in interesting behavior where e.g. both `$PIPELINE.$SECRET` and `$SECRET` are defined, but the network flakes out for the first query, and works for the second, resulting in a query occasionally returning the wrong secret.

Reply:
Sorry, I missed the comment. Will follow up with a better code.